templates/backup.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+# Logic is as follows:
+
+# 1. If pre-script fails exit immediately
+# 2. Proceed if backup
+# 3. Irrespective whether backup succeeds execute post-script.
+
+timestamp=$(date -Isecond)
+
+{% if TARSNAP_BACKUP_PRE_SCRIPT %}
+{{ TARSNAP_BACKUP_PRE_SCRIPT }} || exit 1
+{% endif %}
+
+# We run --fsck before backup, so in case of any inconsistencies (that might happen)
+# backups will still work
+
+/usr/local/bin/tarsnap --fsck --keyfile "{{ TARSNAP_KEY_REMOTE_LOCATION }}" \
```

Comment:
I noted in the cover letter: 

Sometimes backups failed with "Sequence number mismatch: Run --fsck" this could be due to various factors like: 
* me running multiple machines from the same key
* me running backups at the same time they run automatically
* me running backups while provisioning ;). 

Without that any such problem would kill the backups until manual action is performed on server. I added tarsnap --fsck (...) just before backup. This increases backup length (for now increase is acceptable). I believe that in case of backups fail gracefully is sometimes better than fail fast.